src/main/java/org/truffleruby/language/arguments/KeywordArgumentsDescriptor.java

```diff
@@ -16,14 +16,27 @@
  * arguments were passed. The callee must handle that for now. */
 public class KeywordArgumentsDescriptor extends ArgumentsDescriptor {
 
+    public enum SplatType {
+        NO_SPLAT,
+        ONLY_SPLAT,
+        PRE_SPLAT,
+        POST_SPLAT
```

Comment:
Is it useful to have that amount of details? Not against it, but wondering.

What's used for:
```ruby
kw = { b: 2 }
p(a: 1, **kw, c: 3)
# => {:a=>1, :b=>2, :c=>3}
```

Reply:
Pre and post are needed (as the newly added specs show) to know when to override explicit keyword arguments and when not to.

Reply:
One way to simplify (to not have to deal with this complexity on the callee side) is for all kw splats then to let the caller merge into a single RubyHash, and have no literal keywords in the descriptor used for them.
I'm not sure how frequent is a mix of literal and rest kwargs.

Reply:
I think pre and post are easy so I'll try to accommodate them.